usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,259 @@
+            )
+
+        if subawards:
+            return Response(self._handle_subawards(filters))
+
+        return Response(self._handle_awards(filters))
+
+    def _handle_awards(self, filters):
+        """ Award logic for SpendingByAward"""
+        try:
+            obtain_view_from_award_group(filters["award_type_codes"])
+        except Exception:
+
+            error_msg = (
+                '{{"message": "\'award_type_codes\' array can only contain types codes of one group.",'
+                '"award_type_groups": {{'
+                '"contracts": {},'
+                '"loans": {},'
+                '"idvs": {},'
+                '"grants": {},'
+                '"other_financial_assistance": {},'
+                '"direct_payments": {}}}}}'
+            ).format(
+                json.dumps(contract_type_mapping),
+                json.dumps(loan_type_mapping),
+                json.dumps(idv_type_mapping),
+                json.dumps(grant_type_mapping),
+                json.dumps(direct_payment_type_mapping),
+                json.dumps(other_type_mapping),
+            )
+            raise UnprocessableEntityException(json.loads(error_msg))
+
+        awards_values = (
+            list(award_contracts_mapping.keys())
+            + list(loan_award_mapping.keys())
+            + list(non_loan_assistance_award_mapping.keys())
+            + list(award_idv_mapping.keys())
+        )
+
+        if self.sort not in awards_values:
+            raise InvalidParameterException(
+                "Sort value '{}' not found in Award mappings: {}".format(self.sort, awards_values)
+            )
+
+        # build sql query filters
+        model = obtain_view_from_award_group(filters["award_type_codes"])
+        queryset = matview_search_filter(filters, model).values()
+        values = {"award_id", "piid", "fain", "uri", "type"}
+
+        for field in self.fields:
+            if award_contracts_mapping.get(field):
+                values.add(award_contracts_mapping.get(field))
+            if loan_award_mapping.get(field):
+                values.add(loan_award_mapping.get(field))
+            if non_loan_assistance_award_mapping.get(field):
+                values.add(non_loan_assistance_award_mapping.get(field))
+            if award_idv_mapping.get(field):
+                values.add(award_idv_mapping.get(field))
+
+        # Modify queryset to be ordered by requested "sort" in the request or default value(s)
+        if set(filters["award_type_codes"]) <= set(contract_type_mapping):  # contracts
+            sort_filters = [award_contracts_mapping[self.sort]]
+        elif set(filters["award_type_codes"]) <= set(loan_type_mapping):  # loans
+            sort_filters = [loan_award_mapping[self.sort]]
+        elif set(filters["award_type_codes"]) <= set(idv_type_mapping):  # idvs
+            sort_filters = [award_idv_mapping[self.sort]]
+        else:  # assistance data
```

Comment:
```suggestion
        elif set(filters["award_type_codes"]) <= set(assistance_type_mapping): # assistance data
```
I would add a sanity check here in case future changes to the code break this functionality. Having it default to sorting on assistance will be hard to spot in the future

Reply:
I believe this is how it was previously. I wasn't planning to refactor the entire view code even if it now appears to be my intention.

Reply:
reasonable. My suggestion stands, but I certainly won't block on it.